src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/tasks/CleanupEnvelopesDlqTask.java

```diff
@@ -67,18 +66,25 @@ public void deleteMessagesInEnvelopesDlq() throws ServiceBusException, Interrupt
         }
     }
 
-    private void logMessage(IMessage msg) {
-        Envelope envelope = EnvelopeParser.parse(msg.getBody());
-
-        log.info(
-            "Deleting message. ID: {} Envelope ID: {}, File name: {}, Jurisdiction: {}, Classification: {}, Case: {}",
-            msg.getMessageId(),
-            envelope.id,
-            envelope.zipFileName,
-            envelope.jurisdiction,
-            envelope.classification,
-            envelope.caseRef
-        );
+    private void deleteMessage(IMessageReceiver messageReceiver, IMessage message)
+        throws ServiceBusException, InterruptedException {
+        try {
+            JSONObject msgContent = new JSONObject(new String(message.getBody(), defaultCharset()));
+            log.info(
+                "Deleting message. ID: {} "
+                    + "Envelope ID: {}, File name: {}, Jurisdiction: {}, Classification: {}, Case: {}",
+                message.getMessageId(),
+                msgContent.get("id"),
+                msgContent.get("zip_file_name"),
+                msgContent.get("jurisdiction"),
+                msgContent.get("classification"),
+                msgContent.get("case_ref")
+            );
+            messageReceiver.complete(message.getLockToken());
```

Comment:
Just a thought, perhaps better would be to `completeAsync` and then compose the log message, what do you think?